sdk/translation/Azure.AI.Translation.Document/tests/GetAllDocumentsFilterTests.cs

```diff
@@ -0,0 +1,110 @@
+
+            var targetContainer = await CreateTargetContainerAsync();
+            var input = new DocumentTranslationInput(sourceContainer, targetContainer, "fr");
+            var translationOp = await client.StartTranslationAsync(input);
+
+            return translationOp;
+        }
+
+        [RecordedTest]
+        public async Task GetAllDocumentStatusesFilterByStatusTest()
+        {
+            // create client
+            var client = GetClient();
+
+            // create translation job
+            var operation = await CreateSingleTranslationJobAsync(client, docsCount: 3);
+            await operation.WaitForCompletionAsync();
+
+            // list docs
+            var filter = new DocumentFilter(
+                statuses: new List<DocumentTranslationStatus>()
+                {
+                    DocumentTranslationStatus.Succeeded
+                }
+            );
+            var result = operation.GetAllDocumentStatuses(filter: filter);
+
+            // assert.
+            Assert.That(result.All(d => d.Status == DocumentTranslationStatus.Succeeded));
+        }
+
+        [RecordedTest]
+        public async Task GetAllDocumentStatusesFilterByIdsTest()
+        {
+            // create client
+            var client = GetClient();
+
+            // create translation job
+            var operation = await CreateSingleTranslationJobAsync(client, docsCount: 3);
+            await operation.WaitForCompletionAsync();
+            var testIds = operation.GetAllDocumentStatuses().Select(d => d.Id).ToList().GetRange(0, 1);
+
+            // list docs
+            var filter = new DocumentFilter(
+                ids: testIds
+            );
+            var result = operation.GetAllDocumentStatuses(filter: filter);
+
+            // assert
+            Assert.That(result.All(d => testIds.Contains(d.Id)));
+        }
+
+        [Ignore("no way to test this filter")]
+        [RecordedTest]
+        public void GetAllDocumentStatusesFilterByCreatedAfter()
+        {
+        }
+
+        [Ignore("no way to test this filter")]
+        [RecordedTest]
+        public void GetAllDocumentStatusesFilterByCreatedBefore()
+        {
+        }
+
+        [RecordedTest]
+        public async Task GetAllDocumentStatusesOrderByCreatedOn()
+        {
+            // create client
+            var client = GetClient();
+
+            // create translation job
+            var operation = await CreateSingleTranslationJobAsync(client, docsCount: 3);
+            await operation.WaitForCompletionAsync();
+
+            // list docs
+            var filter = new DocumentFilter(
+                orderBy: new List<DocumentFilterOrder> {
+                  new DocumentFilterOrder(property: DocumentFilterProperty.CreatedOn, asc: false)
+                }
+            );
+            var filterResults = operation.GetAllDocumentStatuses(filter: filter);
+
+            // assert
+            var timestamp = Recording.UtcNow;
```

Comment:
the idea of the test is to make sure the output is organized by `CreatedOn` so no need to check what the current time is. You can instead verify that the order is correct. 
For example, that `result[i].CreatedOn < result[i+1].CreatedOn` and so on

Reply:
i'm using it just as a reference point
which will always guaranteed to be greater than the lastly created job